### ISSUE TITLE
Segmentation fault in bfile tests

### ISSUE THREAD

**Issue by hughmcmaster** [indexdata/idzebra#12]
When compiling zebra 2.1.4 on Debian Sid, I'm getting a repeatable segmentation fault (and test failure) in bfile/`tstbfile1`.

```
make  check-TESTS
make[3]: Entering directory '/home/hugh/tmp/idzebra-2.1.4/bfile'
make[4]: Entering directory '/home/hugh/tmp/idzebra-2.1.4/bfile'
../config/test-driver: line 107: 15949 Segmentation fault      "$@" > $log_file 2>&1
FAIL: tstbfile1
PASS: tstmfile1
PASS: tstbfile2
=========================================
   idzebra 2.1.4: bfile/test-suite.log
=========================================

# TOTAL: 3
# PASS:  2
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: tstbfile1
===============

21:24:19-30/03 [log] tstbfile1.c:172 ok: 
21:24:19-30/03 [log] bfs
21:24:19-30/03 [log] tstbfile1.c:37 ok: 
21:24:19-30/03 [log] bf
21:24:20-30/03 [log] tstbfile1.c:45 ok: 
21:24:20-30/03 [log] bf
FAIL tstbfile1 (exit status: 139)
```

I'm not familiar with idzebra's source code, so I don't know how to help here. Can you please investigate?

**Comment by adamdickmeiss** [comment]
Can't reproduce.. I tried debian:sid Docker contain as of today and was not able to see it. Perhaps you can sure config.log, so that I can verify compiler version in use etc?

**Comment by hughmcmaster** [comment]
I've seen it several times. I'll get more information.

**Comment by dltj** [comment]
Hi, @hughmcmaster : just checking in to see if you were able to pin down some more information that would help us reproduce the problem.

**Comment by hughmcmaster** [comment]
@dltj, thanks for reminding me about this. Here are two architectures that failed at the same test I encountered.

An i386 Debian build failed: https://tests.reproducible-builds.org/debian/rb-pkg/bullseye/i386/idzebra.html

A powerpc build failed on Debian: https://buildd.debian.org/status/fetch.php?pkg=idzebra&arch=powerpc&ver=2.1.4-4&stamp=1586186004&raw=0

I need to run more tests on my systems. I've seen this happen on amd64 as well, so I wish I'd investigated further.

**Comment by adamdickmeiss** [comment]
If there's any chance you can try out master that would be great. I have yet to see it failing.

**Comment by adamdickmeiss** [comment]
Zebra 2.2.0 released. Since we were never able to reproduce this issue locally, we'd like to know if it is fixed by this release.

Note that Zebra 2.2.0 may use yaz*pc files for configuration (--with-yaz=pkg) to accomodate for Debian policy of using pkg-config everywhere.

**Comment by hughmcmaster** [comment]
Thanks for the new release. I've started packaging it and hope to release it in a few days.

**Comment by dltj** [comment]
Thank you, @hughmcmaster.

**Comment by adamdickmeiss** [comment]
Re-opened, because I have not heard that the problem is resolved.

**Comment by hughmcmaster** [comment]
After packaging idzebra 2.2.0 for Debian, I built the package 10 times in a clean chroot.

The builds failed twice at the same point - on the tstbfile1 test. I hope these logs help identify the issue.

[idzebra-2.2.0.build1-pass.log](https://github.com/indexdata/idzebra/files/5384630/idzebra-2.2.0.build1-pass.log)
[idzebra-2.2.0.build2-fail.log](https://github.com/indexdata/idzebra/files/5384631/idzebra-2.2.0.build2-fail.log)
[idzebra-2.2.0.build3-pass.log](https://github.com/indexdata/idzebra/files/5384632/idzebra-2.2.0.build3-pass.log)
[idzebra-2.2.0.build4-fail.log](https://github.com/indexdata/idzebra/files/5384633/idzebra-2.2.0.build4-fail.log)
[idzebra-2.2.0.build5-pass.log](https://github.com/indexdata/idzebra/files/5384635/idzebra-2.2.0.build5-pass.log)
[idzebra-2.2.0.build6-pass.log](https://github.com/indexdata/idzebra/files/5384636/idzebra-2.2.0.build6-pass.log)
[idzebra-2.2.0.build7-pass.log](https://github.com/indexdata/idzebra/files/5384637/idzebra-2.2.0.build7-pass.log)
[idzebra-2.2.0.build8-pass.log](https://github.com/indexdata/idzebra/files/5384639/idzebra-2.2.0.build8-pass.log)
[idzebra-2.2.0.build9-pass.log](https://github.com/indexdata/idzebra/files/5384640/idzebra-2.2.0.build9-pass.log)
[idzebra-2.2.0.build10-pass.log](https://github.com/indexdata/idzebra/files/5384641/idzebra-2.2.0.build10-pass.log)

**Comment by adamdickmeiss** [comment]
Solves this by removing the code in question .. Not just the test, but the subsystem that is tested which is not in use anywhere.

**Comment by hughmcmaster** [comment]
That seems best for this issue, particularly if it's not used at all. Thanks.

**Comment by adamdickmeiss** [comment]
Assuming closed by Zebra 2.2.1
